### PR TITLE
Added Goodie Crates

### PR DESCRIPTION
**Pull request by MustaphaTR**
I couldn't get water image work. I need some help for it. Goodie types are just for existance of crate, it should be adjusted for orginal RA2 later.

### CODE REVIEW

File: sequences/misc.yaml [hunk 1]
```diff
@@ -42,16 +42,33 @@ crate-effects: #Temp Null Some SHPs cannot be read by openra at this time, expec
...
+		UseTilesetExtension: true
+	land:
+		Offset: 0, -12
+		ShadowStart: 1
+	water: wcrate
+		ShadowStart: 1
+		Offset: 0, -12
+		UseTilesetExtension: true
+
+wcrate:
```
Comment (Mailaender) [inline sequences/misc.yaml]: This sequence is unused.

File: rules/misc.yaml [hunk 2]
```diff
@@ -16,6 +16,38 @@ waypoint:
 	WithSpriteBody:
 	QuantizeFacingsFromSequence:
 
+CRATE:
```
Comment (Phrohdoh) [inline rules/misc.yaml]: Why `CRATE` and not `crate`?

Comment (MustaphaTR) [inline rules/misc.yaml]: All other mods had `Crate:`. If you want to change it to `crate:`, i'll.

Comment (Phrohdoh) [inline rules/misc.yaml]: Actor definitions in RA2 have all been lowercase so let's go with `crate:`.

**Comment by MustaphaTR** [comment]
Updated but can't squash commits right now.

**Comment by MustaphaTR** [comment]
Squashed.

**Comment by abcdefg30** [comment]
Looks good to me. :+1: / :white_check_mark: 

The crates not showing their water sequence is an upstream issue I'll look into.

**Comment by abcdefg30** [comment]
I filed https://github.com/OpenRA/OpenRA/pull/10568 which makes this work in conjunction with #80.

File: sequences/misc.yaml [hunk 3]
```diff
@@ -42,16 +42,28 @@ crate-effects: #Temp Null Some SHPs cannot be read by openra at this time, expec
...
+	levelup: veteran
 	speed: #speed
 
+crate:
+	idle:
+		Offset: 0, -12
+		UseTilesetExtension: true
+	land:
+		Offset: 0, -12
+		ShadowStart: 1
```
Comment (Mailaender) [inline sequences/misc.yaml]: I think you are missing `UseTilesetExtension: true` here which causes the game to crash:

> Exception of type `System.InvalidOperationException`: Unit `crate` does not have a sequence named `land`

Comment (MustaphaTR) [inline sequences/misc.yaml]: I have no idea why i forgot that. I'll open a PR to fix, or you can do a direct commit without PR.

Comment (Mailaender) [inline sequences/misc.yaml]: It will only crash in conjunction with https://github.com/OpenRA/OpenRA/pull/10568 by the way.

Comment (MustaphaTR) [inline sequences/misc.yaml]: Becuase it uses `idle:` without this pr.

Comment (Mailaender) [inline sequences/misc.yaml]: Can you send us a fixup PR?

Comment (MustaphaTR) [inline sequences/misc.yaml]: Already sent.

**Pull request merged**